File: app/src/main/java/io/github/domi04151309/alwayson/EditIntegerPreference.java
package io.github.domi04151309.alwayson;

import android.content.Context;

import androidx.preference.EditTextPreference;
import android.util.AttributeSet;

public class EditIntegerPreference extends EditTextPreference {

    private Integer mInteger;

    public EditIntegerPreference(Context context, AttributeSet attrs, int defStyle) {
        super(context, attrs, defStyle);
    }

    public EditIntegerPreference(Context context, AttributeSet attrs) {
        super(context, attrs);
    }

    public EditIntegerPreference(Context context) {
        super(context);
    }

    @Override public void setText(String text) {
        final boolean wasBlocking = shouldDisableDependents();
        mInteger = parseInteger(text);
        persistString(mInteger != null ? mInteger.toString() : null);
        final boolean isBlocking = shouldDisableDependents();
        if (isBlocking != wasBlocking) notifyDependencyChange(isBlocking);
    }

    @Override public String getText() {
        return mInteger != null ? mInteger.toString() : null;
    }

    private static Integer parseInteger(String text) {
        try { return Integer.parseInt(text); }
        catch (NumberFormatException e) { return null; }
    }

    @Override protected String getPersistedString(final String defaultReturnValue) {
        int defaultAsInt;
        try {
            defaultAsInt = Integer.parseInt(defaultReturnValue);
        } catch (NumberFormatException e) {
            defaultAsInt = 0;
        }
        final int intValue = getPersistedInt(defaultAsInt);
        return Integer.toString(intValue);
    }

    @Override protected boolean persistString(final String value) {
        try {
            return persistInt(Integer.parseInt(value));
        }catch(NumberFormatException e) {
            return false;
        }
    }
}

File: app/src/main/java/io/github/domi04151309/alwayson/EditIntegerPreference.kt
package io.github.domi04151309.alwayson

import android.content.Context

import androidx.preference.EditTextPreference
import android.util.AttributeSet

class EditIntegerPreference : EditTextPreference {

    constructor(context: Context, attrs: AttributeSet, defStyle: Int) : super(context, attrs, defStyle)

    constructor(context: Context, attrs: AttributeSet) : super(context, attrs)

    constructor(context: Context) : super(context)

    override fun getPersistedString(defaultReturnValue: String?): String {
        val value: Int = if (defaultReturnValue == null) 0 else Integer.parseInt(defaultReturnValue)
        return getPersistedInt(value).toString()
    }

    override fun persistString(value: String): Boolean {
        return persistInt(Integer.parseInt(value))
    }
}